src/Remapping/distributed_remapping.jl

```diff
@@ -777,38 +754,26 @@ function _collect_interpolated_values!(
     if only_one_field
         ClimaComms.reduce!(
```

Comment:
Where is the synchronization of the CUDA stream?

https://github.com/CliMA/ClimaComms.jl/blob/c968350313d5dc81d007721bc27a4bc9c97a79e3/ext/ClimaCommsMPIExt.jl#L69